tests/unit/CompareRaftMembers/DiffStream.js

```diff
@@ -0,0 +1,556 @@
+                    const expectedDiff = [];
+                    for (let i = 0; i < 7777; ++i) {
+                        if (![1234, 6667].includes(i)) {
+                            const paddedI = `000000${i}`.slice(-6);
+                            expectedDiff.push([
+                                null,
+                                { key: `bucket/key-${paddedI}`, value: '{"foo":"bar"}' },
+                            ]);
+                        }
+                    }
+                    return expectedDiff;
+                },
+                withDigests: [
+                    {
+                        desc: 'with no digests DB',
+                        storedDigests: null,
+                        expectedBucketStreamRequests: [
+                            {
+                                bucketName: 'bucket',
+                                marker: null,
+                                lastKey: null,
+                            },
+                        ],
+                    },
+                ],
+            },
+        ].forEach(testCase => {
+            let expectedOutput;
+            if (typeof testCase.expectedOutput === 'function') {
+                expectedOutput = testCase.expectedOutput();
+            } else {
+                expectedOutput = testCase.expectedOutput;
+            }
+            testCase.withDigests.forEach(withDigests => {
+                test(`${testCase.desc} yielding ${expectedOutput.length} diff entries, ${withDigests.desc}`, done => {
+                    if (typeof testCase.bucketdContents === 'function') {
+                        MOCK_BUCKET_STREAM_FULL_LISTING = testCase.bucketdContents();
+                    } else {
+                        MOCK_BUCKET_STREAM_FULL_LISTING = testCase.bucketdContents;
+                    }
+                    let dbContents;
+                    if (typeof testCase.dbContents === 'function') {
+                        dbContents = testCase.dbContents();
+                    } else {
+                        dbContents = testCase.dbContents;
+                    }
+                    const output = [];
+                    const diffStream = new DiffStream({
+                        bucketdHost: 'dummy-host',
+                        bucketdPort: 4242,
+                        maxBufferSize: 2000,
+                        BucketStreamClass: MockBucketStream,
+                    });
+                    diffStream
+                        .on('data', data => {
+                            output.push(data);
+                        })
+                        .on('end', () => {
+                            expect(output).toEqual(expectedOutput);
+                            expect(MOCK_BUCKET_STREAM_REQUESTS_MADE)
+                                .toEqual(withDigests.expectedBucketStreamRequests);
+                            done();
+                        })
+                        .on('error', done);
```

Comment:
if an error occurs before the 'end' event, wouldn't the test appear to succeed here?